src/routes/users.js

```diff
@@ -1,18 +1,20 @@
 import { Router } from 'express';
 import { Permissions } from '../utils';
 import { UserController, RoleController, RequestController } from '../controllers';
-import { AuthMiddleware, RoleMiddleware } from '../middlewares';
+import { AuthMiddleware, RoleMiddleware, TripRequestMiddleware } from '../middlewares';
 
 const router = Router();
 
 const { updateUserRole } = RoleController;
 const { verifyRoles } = RoleMiddleware;
-const { getUserRequests } = RequestController;
+const { getUserRequests, getTripRequestsStats } = RequestController;
 const { userProfile, updateProfile } = UserController;
 const { isAuthenticated, authenticate } = AuthMiddleware;
 const { supplierAdmin } = Permissions;
+const { tripStatsCheck } = TripRequestMiddleware;
 
 router.get('/requests', authenticate, getUserRequests);
+router.post('/request/stats', authenticate, tripStatsCheck, getTripRequestsStats);
```

Comment:
Why is this a POST request and not a GET?

Reply:
I'm using parameters in the body request to query the database. Using `GET` request will throw an error since it does not require passing a body request. I can only implement GET request if the search dates are passed as a query in URL. Oh wait, I think the latter would have been a better approach. Let me refactor it asap